app/controllers/pet_schedules_controller.rb

```diff
@@ -12,7 +12,12 @@ def create
       @schedule.date = Date.parse(date)
       @schedule.save
     end
-    redirect_to pet_path(params[:pet_id])
-    authorize @schedule
+
+    if params[:"booking-dates"].empty?
+      redirect_to new_pet_pet_schedule_path(params[:pet_id]), notice: "Must select available date(s)."
+    else
+      redirect_to pet_path(params[:pet_id])
+    end
+    authorize PetSchedule
```

Comment:
authorize for @schedule is not working here?

Reply:
Yeah, somehow it gave me an error message when I use @schedule. I will try to figure it out tonight.

Reply:
if it works like this should be fine